### DISCUSSION TITLE
如何复现mmgeneration 中的 cyclegan 模型结果

### DISCUSSION THREAD

**Discussion by lovejing0306** [open-mmlab/mmgeneration#465] [How-to] [answered]
你好，我如下配置文件，在 mmgen 中尝试复现你们给出的开源模型结果，但失败了。

我的训练配置信息如下：
配置文件使用是：https://github.com/open-mmlab/mmgeneration/blob/master/configs/cyclegan/cyclegan_lsgan_resnet_in_horse2zebra_b1x1_270k.py

显卡型号：1080 ti
gpu 个数：4

下面是模型训练完之后使用 best 模型在 FID 和 IS 的评估结果
![Xnip2022-03-17_17-59-04](https://user-images.githubusercontent.com/14770340/158786008-ce5fd948-5b27-4b69-adcc-bf7ae3387393.jpg)
下面这个是官方给出的开源模型评估结果
![Xnip2022-03-17_17-59-25](https://user-images.githubusercontent.com/14770340/158786139-8d6a9b8b-7cfb-43ec-a49a-7de3457e3c82.jpg)
从指标上看，自己训练的和官方开源的差异挺大。

请问你们是在什么显卡上用多少个 gpu 训练出了上面的结果？

**Comment by plyfager** [comment]
Hey @lovejing0306  thanks for your interest in our project. 
You can try to train `cyclegan` with 1 GPU. The name field `b1x1_270k` means `batch_size` equals to 1, number of gpu equals to 1.

**Comment by lovejing0306** [comment]
> Hey @lovejing0306 thanks for your interest in our project. You can try to train `cyclegan` with 1 GPU. The name field `b1x1_270k` means `batch_size` equals to 1, number of gpu equals to 1.

ok, thanks. I will try it later .

**Comment by lovejing0306** [comment]
> Hey @lovejing0306 thanks for your interest in our project. You can try to train `cyclegan` with 1 GPU. The name field `b1x1_270k` means `batch_size` equals to 1, number of gpu equals to 1.

 I try to train cyclegan with 1 GPU  and batch size 1 , but the result  is different from yours. 
![Xnip2022-03-19_10-09-01](https://user-images.githubusercontent.com/14770340/159102924-71f4c9d7-7b39-424c-a7d5-5847ed972a54.png)

The best fid is iter_30000, but your best fid is 266800. Can you share your train log ?

**Comment by plyfager** [comment, marked as answer]
I suggest you reduce the eval interval to 100. https://github.com/open-mmlab/mmgeneration/blob/b5ed84d354a25697f48fa9884f0812610a80cedc/configs/cyclegan/cyclegan_lsgan_resnet_in_horse2zebra_b1x1_270k.py#L164

Since the test dataset is very small, the fid results can vary a lot due to different initialization weights. I think the training difference can be ignored.

**Comment by lovejing0306** [comment]
> Since the test dataset is very sma

Thanks for your advice, I will set the interval=100 and try it.
